pylint/testutils/functional/test_file.py

```diff
@@ -0,0 +1,109 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+import configparser
+import sys
+from os.path import basename, exists, join
+from typing import List, Tuple
+
+
+def parse_python_version(ver_str: str) -> Tuple[int, ...]:
+    """Convert python version to a tuple of integers for easy comparison."""
+    return tuple(int(digit) for digit in ver_str.split("."))
+
+
+class NoFileError(Exception):
+    pass
+
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+
+class TestFileOptions(TypedDict):
+    min_pyver: Tuple[int, ...]
+    max_pyver: Tuple[int, ...]
+    min_pyver_end_position: Tuple[int, ...]
+    requires: List[str]
+    except_implementations: str  # Type is actually comma separated list of string
+    exclude_platforms: str  # Type is actually comma separated list of string
```

Comment:
I'll do a follow-up PR where we add this to the `_CONVERTERS` and make this a `List[str]` by default.